coremltools/converters/nnssa/coreml/graph_pass/mlmodel_passes.py

```diff
@@ -244,27 +245,49 @@ def remove_redundant_transposes(spec):
     that compose to the identity.
     """
 
+    def _get_input_output_hashmap(nn_layers):
+        # Hashmap for layers
+        output_to_layers = {}
+        for layer in nn_layers:
+            for input in layer.input:
+                if not input in output_to_layers:
+                    output_to_layers[input] = [layer]
+                else:
+                    output_to_layers[input].append(layer)
+
+        input_to_layer = {}
+        for layer in nn_layers:
+            for output in layer.output:
+                assert(output not in input_to_layer)
```

Comment:
This assert needs to be updated in two ways:

- in Core ML spec a blob with the same name cannot be generated by more than one layers, unless: the layer generating it is of the type "copy". So the check should be only made in the layer type is not `copy`

- update the assert statement to have a useful error message. In general every `assert` should have a useful error message. `assert(output not in input_to_layer, "\'{}\' blob is generated by more than 1 layers".format(output))`

Reply:
good!